smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -307,71 +316,101 @@ def getEcobeeDevices() {
-		]
-		def deviceListParams = [
-				uri:     apiEndpoint,
-				path:    "/1/thermostat",
-				headers: ["Content-Type": "text/json", "Authorization": "Bearer ${state.authToken}"],
-				// TODO - the query string below is not consistent with the Ecobee docs:
-				// https://www.ecobee.com/home/developer/api/documentation/v1/operations/get-thermostats.shtml
-				query: [format: 'json', body: toJson(bodyParams)]
-		]
-		httpGet(deviceListParams) { resp ->
-			if (resp.status == 200) {
-				resp.data.thermostatList.each { stat ->
-					def dni = [app.id, stat.identifier].join('.')
-					thermostatList[dni] = getThermostatDisplayName(stat)
-					// compile all remote sensors conected to the thermostat
-					stat.remoteSensors.each { sensor ->
-							if (sensor.type != "thermostat") {
-								def rsDni = "ecobee_sensor-"+ sensor?.id + "-" + sensor?.code
-								remoteSensors[rsDni] = sensor
-								remoteSensors[rsDni] << [thermostatId: dni]
+				def deviceListParams = [
+						uri:     apiEndpoint,
+						path:    "/1/thermostat",
+						headers: ["Content-Type": "text/json", "Authorization": "Bearer ${state.authToken}"],
+						// TODO - the query string below is not consistent with the Ecobee docs:
+						// https://www.ecobee.com/home/developer/api/documentation/v1/operations/get-thermostats.shtml
+						query: [format: 'json', body: toJson(bodyParams)]
+				]
+				httpGet(deviceListParams) { resp ->
+					isThermostatPolled = true
+					if (resp.status == 200) {
+						resp.data.thermostatList.each { stat ->
+							def dni = [ app.id, stat.identifier ].join('.')
+							def data = getThermostatData(stat)
+							thermostatList[dni] = thermostatList[dni] ? thermostatList[dni] << [data:data] : [data:data]
+							thermostatList[dni].polled = true
+							thermostatList[dni].pollAttempts = 0
+							// compile all remote sensors conected to the thermostat
+							stat.remoteSensors.each { sensor ->
+								if (sensor.type != "thermostat") {
+									def rsDni = "ecobee_sensor-"+ sensor?.id + "-" + sensor?.code
+									remoteSensors[rsDni] = sensor
+									remoteSensors[rsDni] << [thermostatId: dni]
+								}
 							}
+						}
+						state.remoteSensors2 = remoteSensors
+						state.thermostats = thermostatList
+					} else {
+						log.debug "Failed to get thermostats and sensors, status:${resp.status}"
 					}
 				}
-			} else {
-				log.debug "Faile to get thermostats and sensors, status:${resp.status}"
 			}
-		}
-
-		// Now get light swiches
-		def switchListParams = [
-				uri: apiEndpoint + "/ea/devices",
-				headers: ["Content-Type": "application/json;charset=UTF-8", "Authorization": "Bearer ${state.authToken}"],
-		]
-		httpGet(switchListParams) { resp ->
-			if (resp.status == 200) {
-				resp.data?.devices?.each {
-					if (it.type == "LIGHT_SWITCH") {
-						switchList[it?.identifier] = it
-						switchList[it?.identifier] << [deviceAlive: (it?.connected ?: false)]
+			// Now get light swiches
+			if (!isSwitchesPolled) {
```

Comment:
do we need to check this boolean again here? we already have a check above in the while loop.

Reply:
This check might be unecessary as the logic dictates we should at least have been able to poll thermostats before getting here, but it won't prevent the switches to be polled either...